### PR TITLE
Fast up-to-date check support for .editorconfig files

### PR DESCRIPTION
**Pull request by tmeschter**
1. Add EditorConfigFiles.xaml containing an item rule for <EditorConfigFile ... /> MSBuild items. This informs the project system that these are files we need to know about.
2. Add a corresponding EditorConfigFiles.BrowseObject.xaml file so that we can show basic info in the Properties window when clicking on a .editorconfig file in Solution Explorer. Not strictly necessary for fast up-to-date support, but nice to have.
3. Add PropertyPageSchema items for the new .xaml files to Microsoft.Managed.DesignTime.targets. This seems to be necessary for the project system to see that changes have been made in .editorconfig files.
4. Update CSharp.ProjectItemsSchema.xaml and VisualBasic.ProjectItemsSchema.xaml to effectively indicate that EditorConfigFiles items are input source files, like Compile and AdditionalFiles items.

### CODE REVIEW

File: src/Microsoft.VisualStudio.ProjectSystem.Managed/Microsoft.VisualStudio.ProjectSystem.Managed.csproj [hunk 1]
```diff
@@ -385,4 +394,9 @@
     <AdditionalFiles Include="PublicAPI.Shipped.txt" />
     <AdditionalFiles Include="PublicAPI.Unshipped.txt" />
   </ItemGroup>
+  <ItemGroup>
+    <None Update="ProjectSystem\Rules\Items\EditorConfigFiles.BrowseObject.xaml">
```
Comment (davidwengier) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed/Microsoft.VisualStudio.ProjectSystem.Managed.csproj]: You can remove this, its CPS being over-zealous with the project file

File: src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Rules/CSharp.ProjectItemsSchema.xaml [hunk 2]
```diff
@@ -3,15 +3,16 @@
...
+      ItemType="Compile">
+  </ContentType>
 
-    <ItemType Name="Compile" DisplayName="C# compiler"/>
-    <ItemType Name="AdditionalFiles" DisplayName="C# analyzer additional file"/>
-    
-    <FileExtension Name=".cs" ContentType="CSharpFile"/>
+  <ItemType Name="Compile" DisplayName="C# compiler"/>
+  <ItemType Name="AdditionalFiles" DisplayName="C# analyzer additional file"/>
+  <ItemType Name="EditorConfigFiles" DisplayName="C# .editorconfig file" />
```
Comment (davkean) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Rules/CSharp.ProjectItemsSchema.xaml]: I don't think "C# .editorconfig file" is a correct name for this, this is not C# specific at all.

**Comment by davkean** [comment]
This looks correct, but @davidwengier when we do the "delete all the specialized rules" and replace with "AvailableItemName", can we verify that up-to-date still works?

File: src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/DesignTimeTargets/Microsoft.Managed.DesignTime.targets [hunk 3]
```diff
@@ -163,6 +163,14 @@
       <Context>BrowseObject</Context>
     </PropertyPageSchema>
 
+    <PropertyPageSchema Include="$(ManagedXamlNeutralResourcesDirectory)EditorConfigFiles.xaml">
```
Comment (davkean) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/DesignTimeTargets/Microsoft.Managed.DesignTime.targets]: These need to go into C# and VB specific targets. You've only registered them for C# and VB, not for F#.

**Comment by davkean** [comment]
You need to modify setup\Microsoft.VisualStudio.ProjectSystem.Managed.CommonFiles\CommonFiles.swr to add these new files for setup.

File: src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/DesignTimeTargets/Microsoft.Managed.DesignTime.targets (hunk 3 above)
Comment (tmeschter) [reply src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/DesignTimeTargets/Microsoft.Managed.DesignTime.targets]: Fixed.

File: src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Rules/CSharp.ProjectItemsSchema.xaml (hunk 2 above)
Comment (tmeschter) [reply src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Rules/CSharp.ProjectItemsSchema.xaml]: Fixed.

File: src/Microsoft.VisualStudio.ProjectSystem.Managed/Microsoft.VisualStudio.ProjectSystem.Managed.csproj (hunk 1 above)
Comment (tmeschter) [reply src/Microsoft.VisualStudio.ProjectSystem.Managed/Microsoft.VisualStudio.ProjectSystem.Managed.csproj]: Done.

**Comment by tmeschter** [comment]
@davkean I've fixed the setup authoring. Thanks for catching that.

**Pull request merged**